extensions/email/src/main/java/org/mule/extension/email/api/retriever/imap/IMAPConnectionProvider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.api.retriever.imap;
+
+import static org.mule.extension.email.internal.util.EmailConstants.PORT_IMAP;
+import static org.mule.extension.email.internal.util.EmailConstants.PROTOCOL_IMAP;
+import org.mule.extension.email.api.retriever.AbstractRetrieverProvider;
+import org.mule.extension.email.api.retriever.RetrieverConnection;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategy;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategyFactory;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+/**
+ * A {@link ConnectionProvider} that returns instances of imap based {@link RetrieverConnection}s.
+ *
+ * @since 4.0
+ */
+@Alias("imap")
+public class IMAPConnectionProvider extends AbstractRetrieverProvider implements ConnectionProvider<IMAPConfiguration, RetrieverConnection>
+{
+
+    /**
+     * The port number of the mail server. The default value for imap mail servers is 143.
+     */
+    @Parameter
+    @Optional(defaultValue = PORT_IMAP)
+    private String port;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RetrieverConnection connect(IMAPConfiguration config) throws ConnectionException
+    {
+        return new RetrieverConnection(PROTOCOL_IMAP, getUser(), getPassword(), getHost(), getPort(), getProperties(), null, config.getFolder());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnect(RetrieverConnection connection)
+    {
+        connection.disconnect();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ConnectionValidationResult validate(RetrieverConnection connection)
+    {
+        return ConnectionValidationResult.success();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ConnectionHandlingStrategy<RetrieverConnection> getHandlingStrategy(ConnectionHandlingStrategyFactory<IMAPConfiguration, RetrieverConnection> connectionHandlingStrategyFactory)
+    {
+        return connectionHandlingStrategyFactory.supportsPooling();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getPort()
```

Comment:
what is this needed for? who will consume it? Shouldn't the provider just set the port on the connection and that's it?